### PR TITLE
build: Transition from Java EE to Jakarta EE

### PR DESCRIPTION
**Pull request by skabashnyuk**
### What does this PR do?
The transition from Java EE to Jakarta EE
PR can be reviewed commit by commit.
TODO 
Everrest CQ - https://dev.eclipse.org/ipzilla/show_bug.cgi?id=23669

### Screenshot/screencast of this PR
Знімок екрана 2021-08-10 о 16 28 12
Знімок екрана 2021-08-10 о 16 27 36

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/20247

### How to test this PR?
-  build or deploy che-server's image `quay.io/skabashn/che-server:jakarta_2021_09_03_14_12_20`
- login as "kubeadmin" on OpenShift
- Start new workspace.
- Workspace should start successfully.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [x] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Comment by dmytro-ndp** [comment]
[crw-ci-test --rebuild]

**Comment by dmytro-ndp** [comment]
[crw-ci-test --rebuild]

**Comment by sparkoo** [comment]
I can see following error in che-server boot logs. Everything seems to work though.

```
2021-08-23 11:19:16,150[main]             [DEBUG] [o.a.c.c.C.[.[localhost].[/api] 175]  - Failed to load class [com.fasterxml.jackson.jaxrs.json.JacksonJaxbJsonProvider]
java.lang.NoClassDefFoundError: javax/ws/rs/ext/MessageBodyReader
	at java.base/java.lang.ClassLoader.defineClass1(Native Method)
	at java.base/java.lang.ClassLoader.defineClass(Unknown Source)
	at java.base/java.security.SecureClassLoader.defineClass(Unknown Source)
	at org.apache.catalina.loader.WebappClassLoaderBase.findClassInternal(WebappClassLoaderBase.java:2516)
	at org.apache.catalina.loader.WebappClassLoaderBase.findClass(WebappClassLoaderBase.java:872)
	at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1408)
	at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1252)
	at java.base/java.lang.ClassLoader.defineClass1(Native Method)
	at java.base/java.lang.ClassLoader.defineClass(Unknown Source)
	at java.base/java.security.SecureClassLoader.defineClass(Unknown Source)
	at org.apache.catalina.loader.WebappClassLoaderBase.findClassInternal(WebappClassLoaderBase.java:2516)
	at org.apache.catalina.loader.WebappClassLoaderBase.findClass(WebappClassLoaderBase.java:872)
	at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1408)
	at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1252)
	at java.base/java.lang.ClassLoader.defineClass1(Native Method)
	at java.base/java.lang.ClassLoader.defineClass(Unknown Source)
	at java.base/java.security.SecureClassLoader.defineClass(Unknown Source)
	at org.apache.catalina.loader.WebappClassLoaderBase.findClassInternal(WebappClassLoaderBase.java:2516)
	at org.apache.catalina.loader.WebappClassLoaderBase.findClass(WebappClassLoaderBase.java:872)
	at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1408)
	at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1252)
	at org.apache.catalina.util.Introspection.loadClass(Introspection.java:141)
	at org.apache.catalina.startup.ContextConfig.checkHandlesTypes(ContextConfig.java:2440)
	at org.apache.catalina.startup.ContextConfig.processAnnotationsStream(ContextConfig.java:2342)
	at org.apache.catalina.startup.ContextConfig.processAnnotationsJar(ContextConfig.java:2294)
	at org.apache.catalina.startup.ContextConfig.processAnnotationsUrl(ContextConfig.java:2264)
	at org.apache.catalina.startup.ContextConfig.scanWebXmlFragment(ContextConfig.java:2172)
	at org.apache.catalina.startup.ContextConfig.processAnnotations(ContextConfig.java:2152)
	at org.apache.catalina.startup.ContextConfig.processClasses(ContextConfig.java:1408)
	at org.apache.catalina.startup.ContextConfig.webConfig(ContextConfig.java:1303)
	at org.apache.catalina.startup.ContextConfig.configureStart(ContextConfig.java:986)
	at org.apache.catalina.startup.ContextConfig.lifecycleEvent(ContextConfig.java:303)
	at org.apache.catalina.util.LifecycleBase.fireLifecycleEvent(LifecycleBase.java:123)
	at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5084)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
	at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:726)
	at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:698)
	at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:747)
	at org.apache.catalina.startup.HostConfig.deployWAR(HostConfig.java:1027)
	at org.apache.catalina.startup.HostConfig$DeployWar.run(HostConfig.java:2001)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at org.apache.tomcat.util.threads.InlineExecutorService.execute(InlineExecutorService.java:75)
	at java.base/java.util.concurrent.AbstractExecutorService.submit(Unknown Source)
	at org.apache.catalina.startup.HostConfig.deployWARs(HostConfig.java:828)
	at org.apache.catalina.startup.HostConfig.deployApps(HostConfig.java:478)
	at org.apache.catalina.startup.HostConfig.start(HostConfig.java:1708)
	at org.apache.catalina.startup.HostConfig.lifecycleEvent(HostConfig.java:320)
	at org.apache.catalina.util.LifecycleBase.fireLifecycleEvent(LifecycleBase.java:123)
	at org.apache.catalina.util.LifecycleBase.setStateInternal(LifecycleBase.java:423)
	at org.apache.catalina.util.LifecycleBase.setState(LifecycleBase.java:366)
	at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:946)
	at org.apache.catalina.core.StandardHost.startInternal(StandardHost.java:886)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1396)
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1386)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at org.apache.tomcat.util.threads.InlineExecutorService.execute(InlineExecutorService.java:75)
	at java.base/java.util.concurrent.AbstractExecutorService.submit(Unknown Source)
	at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:919)
	at org.apache.catalina.core.StandardEngine.startInternal(StandardEngine.java:263)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
	at org.apache.catalina.core.StandardService.startInternal(StandardService.java:432)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
	at org.apache.catalina.core.StandardServer.startInternal(StandardServer.java:927)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
	at org.apache.catalina.startup.Catalina.start(Catalina.java:795)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.apache.catalina.startup.Bootstrap.start(Bootstrap.java:345)
	at org.apache.catalina.startup.Bootstrap.main(Bootstrap.java:476)
Caused by: java.lang.ClassNotFoundException: javax.ws.rs.ext.MessageBodyReader
	at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1444)
	at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1252)
	... 73 common frames omitted
2021-08-23 11:19:16,154[main]             [DEBUG] [o.a.c.c.C.[.[localhost].[/api] 175]  - Failed to load class [com.fasterxml.jackson.jaxrs.json.JacksonJsonProvider]
java.lang.NoClassDefFoundError: javax/ws/rs/ext/MessageBodyReader
	at java.base/java.lang.ClassLoader.defineClass1(Native Method)
	at java.base/java.lang.ClassLoader.defineClass(Unknown Source)
	at java.base/java.security.SecureClassLoader.defineClass(Unknown Source)
	at org.apache.catalina.loader.WebappClassLoaderBase.findClassInternal(WebappClassLoaderBase.java:2516)
	at org.apache.catalina.loader.WebappClassLoaderBase.findClass(WebappClassLoaderBase.java:872)
	at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1408)
	at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1252)
	at java.base/java.lang.ClassLoader.defineClass1(Native Method)
	at java.base/java.lang.ClassLoader.defineClass(Unknown Source)
	at java.base/java.security.SecureClassLoader.defineClass(Unknown Source)
	at org.apache.catalina.loader.WebappClassLoaderBase.findClassInternal(WebappClassLoaderBase.java:2516)
	at org.apache.catalina.loader.WebappClassLoaderBase.findClass(WebappClassLoaderBase.java:872)
	at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1408)
	at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1252)
	at org.apache.catalina.util.Introspection.loadClass(Introspection.java:141)
	at org.apache.catalina.startup.ContextConfig.checkHandlesTypes(ContextConfig.java:2440)
	at org.apache.catalina.startup.ContextConfig.processAnnotationsStream(ContextConfig.java:2342)
	at org.apache.catalina.startup.ContextConfig.processAnnotationsJar(ContextConfig.java:2294)
	at org.apache.catalina.startup.ContextConfig.processAnnotationsUrl(ContextConfig.java:2264)
	at org.apache.catalina.startup.ContextConfig.scanWebXmlFragment(ContextConfig.java:2172)
	at org.apache.catalina.startup.ContextConfig.processAnnotations(ContextConfig.java:2152)
	at org.apache.catalina.startup.ContextConfig.processClasses(ContextConfig.java:1408)
	at org.apache.catalina.startup.ContextConfig.webConfig(ContextConfig.java:1303)
	at org.apache.catalina.startup.ContextConfig.configureStart(ContextConfig.java:986)
	at org.apache.catalina.startup.ContextConfig.lifecycleEvent(ContextConfig.java:303)
	at org.apache.catalina.util.LifecycleBase.fireLifecycleEvent(LifecycleBase.java:123)
	at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5084)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
	at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:726)
	at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:698)
	at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:747)
	at org.apache.catalina.startup.HostConfig.deployWAR(HostConfig.java:1027)
	at org.apache.catalina.startup.HostConfig$DeployWar.run(HostConfig.java:2001)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at org.apache.tomcat.util.threads.InlineExecutorService.execute(InlineExecutorService.java:75)
	at java.base/java.util.concurrent.AbstractExecutorService.submit(Unknown Source)
	at org.apache.catalina.startup.HostConfig.deployWARs(HostConfig.java:828)
	at org.apache.catalina.startup.HostConfig.deployApps(HostConfig.java:478)
	at org.apache.catalina.startup.HostConfig.start(HostConfig.java:1708)
	at org.apache.catalina.startup.HostConfig.lifecycleEvent(HostConfig.java:320)
	at org.apache.catalina.util.LifecycleBase.fireLifecycleEvent(LifecycleBase.java:123)
	at org.apache.catalina.util.LifecycleBase.setStateInternal(LifecycleBase.java:423)
	at org.apache.catalina.util.LifecycleBase.setState(LifecycleBase.java:366)
	at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:946)
	at org.apache.catalina.core.StandardHost.startInternal(StandardHost.java:886)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1396)
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1386)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at org.apache.tomcat.util.threads.InlineExecutorService.execute(InlineExecutorService.java:75)
	at java.base/java.util.concurrent.AbstractExecutorService.submit(Unknown Source)
	at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:919)
	at org.apache.catalina.core.StandardEngine.startInternal(StandardEngine.java:263)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
	at org.apache.catalina.core.StandardService.startInternal(StandardService.java:432)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
	at org.apache.catalina.core.StandardServer.startInternal(StandardServer.java:927)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
	at org.apache.catalina.startup.Catalina.start(Catalina.java:795)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.apache.catalina.startup.Bootstrap.start(Bootstrap.java:345)
	at org.apache.catalina.startup.Bootstrap.main(Bootstrap.java:476)
Caused by: java.lang.ClassNotFoundException: javax.ws.rs.ext.MessageBodyReader
	at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1444)
	at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1252)
	... 66 common frames omitted
```

**Comment by skabashnyuk** [comment]
>I can see following error in che-server boot logs. Everything seems to work though. I'm testing now with devworkspaces.

Didn't test that on devworkspaces, only che7 based workspaces. Let me look at that.

**Comment by sparkoo** [comment]
I can see it with both dw enabled (https://gist.githubusercontent.com/sparkoo/6a461eee24ae2a7770dbaffd37750046/raw/a02557018e36640ce3a8b3cf17229e6c76133203/dw.log) and disabled (https://gist.githubusercontent.com/sparkoo/6a461eee24ae2a7770dbaffd37750046/raw/a02557018e36640ce3a8b3cf17229e6c76133203/nodw.log), but I've noticed it's on DEBUG level

**Comment by sparkoo** [comment]
I can't see the error in che-server:next

File: multiuser/machine-auth/che-multiuser-machine-authentication/src/test/java/org/eclipse/che/multiuser/machine/authentication/server/MachineTokenRegistryTest.java [hunk 1]
```diff
@@ -53,7 +53,7 @@
 
   private MachineTokenRegistry tokenRegistry;
 
-  private static final int KEY_SIZE = 1024;
+  private static final int KEY_SIZE = 2048;
```
Comment (sparkoo) [inline multiuser/machine-auth/che-multiuser-machine-authentication/src/test/java/org/eclipse/che/multiuser/machine/authentication/server/MachineTokenRegistryTest.java:56]: is it related?

**Review by metlos — COMMENTED** [review]
I agree with @sparkoo 's comments and have some more minor things...

**Comment by skabashnyuk** [comment]
[crw-ci-test --rebuild]

**Comment by skabashnyuk** [comment]
[crw-ci-test --rebuild]

**Comment by skabashnyuk** [comment]
@sparkoo @metlos I've addressed your comments. Code tested with che7 and devworkspaces.

**Comment by skabashnyuk** [comment]
[crw-ci-test --rebuild]

**Review by sparkoo — APPROVED** [review]
tested with devworkspaces

**Comment by skabashnyuk** [comment]
[crw-ci-test --rebuild]

**Pull request merged**